server/base/src/main/java/org/apache/accumulo/server/conf/store/impl/PropStoreWatcher.java

```diff
@@ -0,0 +1,253 @@
+   * the change - if they call get to update and respond to the change the watcher will be (re)set
+   * then. This helps clean up watchers by not automatically re-adding the watcher on the event but
+   * only if being used.
+   *
+   * @param event
+   *          ZooKeeper event.
+   */
+  @SuppressWarnings("FutureReturnValueIgnored") // currently, tasks are fire and forget
+  @Override
+  public void process(final WatchedEvent event) {
+
+    String path;
+    switch (event.getType()) {
+      case NodeDataChanged:
+        path = event.getPath();
+        log.trace("handle change event for path: {}", path);
+        PropCacheId.fromPath(path).ifPresent(this::signalZkChangeEvent);
+        break;
+      case NodeDeleted:
+        path = event.getPath();
+        log.trace("handle delete event for path: {}", path);
+        PropCacheId.fromPath(path).ifPresent(cacheId -> {
+          // notify listeners
+          Set<PropChangeListener> snapshot = getListenerSnapshot(cacheId);
+          if (snapshot != null) {
+            executorService
+                .submit(new PropStoreEventTask.PropStoreDeleteEventTask(cacheId, snapshot));
+          }
+
+          listenerCleanup(cacheId);
+
+        });
+
+        break;
+      case None:
+        Event.KeeperState state = event.getState();
+        switch (state) {
+          // pause - could reconnect
+          case ConnectedReadOnly:
+          case Disconnected:
+            log.debug("ZooKeeper disconnected event received");
+            zkReadyMonitor.clearReady();
+            executorService.submit(new PropStoreEventTask.PropStoreConnectionEventTask(null,
+                getAllListenersSnapshot()));
+            break;
+
+          // okay
+          case SyncConnected:
+            log.debug("ZooKeeper connected event received");
+            zkReadyMonitor.setReady();
+            break;
+
+          // terminal - never coming back.
+          case Expired:
+          case Closed:
+            log.info("ZooKeeper connection closed event received");
+            zkReadyMonitor.clearReady();
+            zkReadyMonitor.setClosed(); // terminal condition
+            executorService.submit(new PropStoreEventTask.PropStoreConnectionEventTask(null,
+                getAllListenersSnapshot()));
+            break;
+
+          default:
+            log.trace("ignoring zooKeeper state: {}", state);
+        }
+        break;
+      default:
+        break;
+    }
+
+  }
+
+  /**
+   * Submit task to notify registered listeners that the propCacheId node received an event
+   * notification from ZooKeeper and should be updated. The process can be initiated either by a
+   * ZooKeeper notification or a change detected in the cache based on a ZooKeeper event.
+   *
+   * @param propCacheId
+   *          the cache id
+   */
+  @SuppressWarnings("FutureReturnValueIgnored") // currently, tasks are fire and forget
```

Comment:
Suppressing non-standard warnings can generate new warnings in compilers/IDEs that don't recognize these non-standard ones (for good reason: unrecognized warnings submissions could be typos and not what the user intends). The ignoring of the return value should be solved in a different way, such as calling execute instead of submit.

Reply:
fixed in e7dc935faf